src/htm/algorithms/Connections.cpp

```diff
@@ -250,18 +250,14 @@ void Connections::destroySynapse(const Synapse synapse) {
     }
   }
 
-  const auto synapseOnSegment =
-      std::lower_bound(segmentData.synapses.cbegin(), segmentData.synapses.cend(),
-                       synapse, 
-		       [&](const Synapse a, const Synapse b) -> bool {
-			 return dataForSynapse(a).id < dataForSynapse(b).id;
-                       });
-
-  NTA_ASSERT(synapseOnSegment != segmentData.synapses.end());
+  const auto synapseOnSegment = std::find(segmentData.synapses.cbegin(), 
```

Comment:
cherry-picked this fix from another PR

Reply:
why doesn't `lower_bound` work anymore?  In theory it should be a lot faster than `find`...

Reply:
I'll try reverting the commit..maybe the synapses were not sorted after pruning (well, that would not change that) 

>  In theory it should be a lot faster than find...

yes, O(log(N)), vs O(N) but irl for such small vector there does not seem to be any difference.

Reply:
reverted back to lower_b and its form.